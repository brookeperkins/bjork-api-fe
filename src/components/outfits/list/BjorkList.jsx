import React from 'react';
import { useOutfits } from '../../../hooks/useOutfits';
import BjorkItem from './BjorkItem';
import { Link } from 'react-router-dom';

const BjorkList = () => {
  const { loading, outfits } = useOutfits();
  console.log(outfits);

  if(loading) return <h1>Loading...</h1>;

  const bjorkElements = outfits.map(outfit => (
    <li key={outfit.id}>
      <Link to={`/${outfit.id}`}>
        <BjorkItem {...outfit} />
      </Link>
    </li>
  ));

  return (
    <figure>
      <ul data-testid="bjork-list">
        {bjorkElements}
      </ul>
    </figure>
  );
};

export default BjorkList;
