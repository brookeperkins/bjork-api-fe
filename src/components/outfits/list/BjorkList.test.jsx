import React from 'react';
import { render, cleanup } from '@testing-library/react';
import BjorkList from './BjorkList';
/* eslint-disable max-len */

const testArray = [
  { img: 'Bjork.jpg', 
    year: 1999 },
  { img: 'Bjork2.jpg', 
    year: 2018 }
];

describe('<BjorkList />', () => {
  afterEach(() => cleanup());
  it('renders the BjorkList component', () => {
    const { asFragment } = render(
      <BjorkList
        outfits={testArray}
      />);
    expect(asFragment()).toMatchSnapshot();
  });
});

