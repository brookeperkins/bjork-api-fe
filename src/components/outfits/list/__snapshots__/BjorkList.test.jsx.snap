// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BjorkList /> renders the BjorkList component 1`] = `
<DocumentFragment>
  <h1>
    Loading...
  </h1>
</DocumentFragment>
`;
